src/overlays/actors/ovl_En_Dekunuts/z_en_dekunuts.c

```diff
@@ -81,97 +122,635 @@ static DamageTable D_808BEF64 = {
+}
+
+void func_808BDA08(EnDekunuts* this) {
+    SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, &D_06000168);
+    this->unk_190 = this->unk_194;
+    this->actionFunc = func_808BDA4C;
+}
+
+void func_808BDA4C(EnDekunuts* this, GlobalContext* globalCtx) {
+    Player* player;
+    Vec3f sp58;
+    s16 pitch;
+    Vec3f pos;
+    f32 val;
+    s16 params;
+
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 2, 0xE38);
+    if (this->actor.params == ENDEKUNUTS_PARAMS_2) {
+        player = PLAYER;
+
+        sp58.x = player->actor.world.pos.x;
+        sp58.z = player->actor.world.pos.z;
+        sp58.y = player->actor.world.pos.y + 40.0f;
+        pitch = Actor_PitchToPoint(&this->actor, &sp58);
+        pitch = CLAMP(pitch, -0x3800, -0x2000);
+        if (this->skelAnime.animCurrentFrame < 7.0f) {
+            Math_ScaledStepToS(&this->actor.world.rot.x, pitch, 0x800);
+        } else {
+            Math_ScaledStepToS(&this->actor.world.rot.x, 0, 0x800);
+        }
+    }
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        this->actor.world.rot.x = 0;
+        func_808BD870(this);
+    } else if (func_801378B8(&this->skelAnime, 7.0f)) {
+        val = Math_CosS(this->actor.world.rot.x) * 15.0f;
+        pos.x = (Math_SinS(this->actor.shape.rot.y) * val) + this->actor.world.pos.x;
+        pos.y = (this->actor.world.pos.y + 12.0f) - (Math_SinS(this->actor.world.rot.x) * 15.0f);
+        pos.z = (Math_CosS(this->actor.shape.rot.y) * val) + this->actor.world.pos.z;
+        params = (this->actor.params == ENDEKUNUTS_PARAMS_2) ? ENDEKUNUTS_PARAMS_2 : ENDEKUNUTS_PARAMS_0;
+
+        if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_NUTSBALL, pos.x, pos.y, pos.z,
+                        this->actor.world.rot.x, this->actor.shape.rot.y, 0, params) != NULL) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_THROW);
+        }
+    } else if ((this->unk_190 >= 2) && func_801378B8(&this->skelAnime, 12.0f)) {
+        SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_06000168, -3.0f);
+        if (this->unk_190 != 0) {
+            this->unk_190--;
+        }
+    }
+}
+
+void func_808BDC9C(EnDekunuts* this) {
+    SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_06002A5C, -5.0f);
+    this->unk_190 = 0;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_DOWN);
+    this->actionFunc = func_808BDD54;
+}
+
+void func_808BDCF0(EnDekunuts* this) {
+    SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_06002A5C, -5.0f);
+    this->collider.base.acFlags &= ~AC_ON;
+    this->unk_190 = 80;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_DOWN);
+    this->actionFunc = func_808BDD54;
+}
+
+void func_808BDD54(EnDekunuts* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        if (this->unk_190 != 0) {
+            this->unk_190--;
+        }
+
+        if (this->unk_190 == 0) {
+            func_808BD428(this);
+        }
```

Comment:
I wonder if the `DECR` macro could work here
```suggestion
        if (DECR(this->unk_190) == 0) {
            func_808BD428(this);
        }
```

Reply:
Unfortunately not, adds a whole extra branch in.